gno.land/cmd/gnoland/config.go

```diff
@@ -67,24 +68,83 @@ func getFieldAtPath(currentValue reflect.Value, path []string) (*reflect.Value,
 	return &field, nil
 }
 
-// newInvalidFieldError creates an error for non-existent struct fields
-// being passed as arguments to [getFieldAtPath]
-func newInvalidFieldError(field string, value reflect.Value) error {
-	var (
-		valueType = value.Type()
-		numFields = value.NumField()
-	)
-
-	fields := make([]string, 0, numFields)
+func fieldByTOMLName(value reflect.Value, name string) (dst reflect.Value) {
+	eachTOMLField(value, func(val reflect.Value, tomlName string) bool {
+		if tomlName == name {
+			dst = val
+			return true
+		}
+		return false
+	})
+	return
+}
 
-	for i := 0; i < numFields; i++ {
-		valueField := valueType.Field(i)
-		if !valueField.IsExported() {
+// eachTOMLField iterates over each field in value (assumed to be a struct).
+// For every field within the struct, iterationCallback is called with the value
+// of the field and the associated name in the TOML representation
+// (through the `toml:"..."` struct field, or just the field's name as fallback).
+// If iterationCallback returns true, the function returns immediately with
+// true. If it always returns false, or no fields were processed, eachTOMLField
+// will return false.
+func eachTOMLField(value reflect.Value, iterationCallback func(val reflect.Value, tomlName string) bool) bool {
```

Comment:
We never utilize the return value of this function

It's weird that it returns a boolean at all, considering it takes in a callback that has a range declaration (returns a boolean)

Reply:
We actually do!

It's used recursively within the function itself, so that if we call it on an embedded struct we know if a value was found and we should stop execution.

Reply:
Ah, I am actually blind
![1635886853456](https://github.com/gnolang/gno/assets/16712663/a2bfc30a-779d-458b-bf18-a5fd527535d4)